modelarmor/create_template.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Sample code for creating a new model armor template.
+
+package modelarmor
+
+// [START modelarmor_create_template]
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	modelarmor "cloud.google.com/go/modelarmor/apiv1"
+	modelarmorpb "cloud.google.com/go/modelarmor/apiv1/modelarmorpb"
+	"google.golang.org/api/option"
+)
+
+// createModelArmorTemplate creates a new Model Armor template.
+//
+// This method creates a new Model Armor template with the provided settings.
+//
+// Args:
+//
+//	w io.Writer: The writer to use for logging.
+//	projectID string: The ID of the Google Cloud project.
+//	locationID string: The ID of the Google Cloud location.
+//	templateID string: The ID of the template to create.
+//
+// Returns:
+//
+//	*modelarmorpb.Template: The created template.
+//	error: Any error that occurred during template creation.
+//
+// Example:
+//
+//	template, err := createModelArmorTemplate(
+//	    os.Stdout,
+//	    "my-project",
+//	    "us-central1",
+//	    "my-template",
+//	)
+//	if err != nil {
+//	    log.Fatal(err)
+//	}
+//	fmt.Println(template)
+func createModelArmorTemplate(w io.Writer, projectID, location, templateID string) (*modelarmorpb.Template, error) {
+	ctx := context.Background()
+
+	// Create the Model Armor client.
+	client, err := modelarmor.NewClient(ctx,
+		option.WithEndpoint(fmt.Sprintf("modelarmor.%s.rep.googleapis.com:443", location)),
```

Comment:
issue: instantiate the `CallOptions` object outside of the `NewClient()` call.

Suggestion:

```go
opts := option.WithEndpoint( /* etc */ )
client, err := modelarmor.NewClient(ctx, opts)
// etc
```